xbmc/cores/RetroPlayer/cheevos/Cheevos.cpp

```diff
@@ -126,6 +156,28 @@ void CCheevos::EnableRichPresence()
   m_richPresenceScript.clear();
 }
 
+void CCheevos::ActivateAchievement()
+{
+  if (m_activatedCheevoMap.empty())
+  {
+    if (!LoadData())
+    {
+      CLog::Log(LOGERROR, "Cheevos: Couldn't load patch file");
+      return;
+    }
+    else
+    {
+      CLog::Log(LOGERROR, "No active core achievement for the game");
+    }
+  }
+  for (auto& it : m_activatedCheevoMap)
```

Comment:
something cool we can do with c++17 is structured binding. This get rid of the ugly use of `first/second` in maps.

```suggestion
  for (auto& [id,  Strings] : m_activatedCheevoMap)
```

then later you can use
```c++
    m_gameClient->Cheevos().ActivateAchievement(id, Strings[0].c_str());
```